huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_appcode_test.go

```diff
@@ -0,0 +1,155 @@
+package apig
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk/openstack/apigw/dedicated/v2/applications"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getAppcodeFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := cfg.ApigV2Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating APIG v2 client: %s", err)
+	}
+	return applications.GetAppCode(client, state.Primary.Attributes["instance_id"],
+		state.Primary.Attributes["application_id"], state.Primary.ID).Extract()
+}
+
+// Auto generate APPCODE.
+func TestAccAppcode_basic(t *testing.T) {
+	var (
+		appCode applications.AppCode
+
+		rName = "huaweicloud_apig_appcode.test"
+		rc    = acceptance.InitResourceCheck(rName, &appCode, getAppcodeFunc)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAppcode_basic(),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+				),
+			},
+			{
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccAppcodeImportIdFunc(),
+			},
+		},
+	})
+}
+
+func testAccAppcodeImportIdFunc() resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		rName := "huaweicloud_apig_appcode.test"
+		rs, ok := s.RootModule().Resources[rName]
+		if !ok {
+			return "", fmt.Errorf("Resource (%s) not found: %s", rName, rs)
+		}
+		if rs.Primary.ID == "" || rs.Primary.Attributes["instance_id"] == "" || rs.Primary.Attributes["application_id"] == "" {
+			return "", fmt.Errorf("invalid format specified for import ID, want '<instance_id>/<application_id>/<policy_id>', but got '%s/%s/%s'",
+				rs.Primary.Attributes["instance_id"], rs.Primary.Attributes["application_id"], rs.Primary.ID)
+		}
+		return fmt.Sprintf("%s/%s/%s", rs.Primary.Attributes["instance_id"], rs.Primary.Attributes["application_id"], rs.Primary.ID), nil
```

Comment:
Define variables for repeatedly referenced attributes.
```
id := rs.Primary.ID
instanceId := rs.Primary.Attributes["instance_id"]
applicationId := rs.Primary.Attributes["application_id"]
```

Reply:
The log message `<policy_id>` should be `<id>`

Reply:
fixed.